### ISSUE TITLE
4D point layers show `n-1` 3D slices instead of `n`

### ISSUE THREAD

**Issue by brisvag** [napari/napari#1797]
## 🐛 Bug

When loading a set of 4D coordinates in a points layer, the last "index" of the 4th dimension is not accessible through the slider at the bottom of the viewer. However, if something else forces napari to display that missing slice, the data in that slice is displayed properly.

## To Reproduce
`$ ipython --gui=qt`
```
import numpy as np
import napari

v = napari.Viewer(ndisplay=3)
# 2 points in each of three 3D slices, spaced out differently to easily spot them
points = np.array([
[0,0,0,0],
[0,1,1,1],
[1,0,0,0],
[1,5,5,5],
[2,0,0,0],
[2,10,10,10],
])
v.add_points(points) # This displays only 2 slices. The last two points are missing!
```
This will hold true no matter how long the 4th dimension is. However, if to the earlier code we add:
```
image = np.random.rand(3,10,10,10)
v.add_image(image)
```
then the 3rd slice is recovered. this won't happen if the image is just 1 shorter, like with `image = np.random.rand(2,10,10,10)`.

## Expected behavior

The slider allows access to the whole array, including the last 3D slice.

## Environment

```
napari: 0.3.8
Platform: Linux-5.9.1-arch1-1-x86_64-with-glibc2.2.5
Python: 3.8.6 (default, Sep 30 2020, 04:00:38) [GCC 10.2.0]
Qt: 5.14.2
PyQt5: 5.14.2
NumPy: 1.19.2
SciPy: 1.5.3
Dask: 2.30.0
VisPy: 0.6.5

GL version: 4.6.0 NVIDIA 455.28
MAX_TEXTURE_SIZE: 16384

Plugins:
- mrcfile-reader: 0.1.2
- napari_plugin_engine: 0.1.8
- svg: 0.1.4
```
@alisterburt

**Comment by sofroniewn** [comment]
Hi @brisvag, yes, sorry about this! This should be fixed in the 0.4.0 release which came out yesterday. I think it was fixed by #1696 and was a bug in the calculation of our layer extents.

If I try this right now on 0.3.8 I see what you saw, only two slices, as follows

![4D_points_slice_v038](https://user-images.githubusercontent.com/6531703/97764390-b9a1bd00-1acb-11eb-8230-4b6ec0ce5c2e.gif)

But now on 0.4.0 I see all three slices, as follows

![4D_points_slice](https://user-images.githubusercontent.com/6531703/97764399-c9b99c80-1acb-11eb-978a-ea24422ea8e5.gif)

If you can upgrade to the latest version and try this out it would be much appreciated. Please let us know though if this isn't fixed for you!!

**Comment by brisvag** [comment]
Thank you, I can confirm that updating to 0.4.0 fixed the problem for me as well!
